addons/controllers/cluster_metadata_controller.go

```diff
@@ -92,7 +92,7 @@ func (r *ClusterMetadataReconciler) Reconcile(ctx context.Context, req reconcile
 		}
 	}
 
-	tkr, err := util.GetTKRByNameV1Alpha3(r.context, r.Client, tkrName)
+	tkr, err := util.GetTKRByNameV1Alpha3(r.context, r.Client, cluster, tkrName)
```

Comment:
cluster will never be nil. so why pass it in just to check if it is nil?

Reply:
we are need the annotations in the cluster hence we are passing the cluster here